File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/BotDeathEvent.java
package dev.robocode.tankroyale.botapi.events;

/** Event occurring when a bot has died */
@SuppressWarnings("unused")
public final class BotDeathEvent extends Event {

  /** ID of the bot that has died */
  private final int victimId;

  public BotDeathEvent(int turnNumber, int victimId) {
    super(turnNumber);
    this.victimId = victimId;
  }

  /** Returns the ID of the bot that has died */
  public int getVictimId() {
    return victimId;
  }
}

File: bot-api/dotnet/Robocode.TankRoyale.BotApi/src/events/BotDeathEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when another bot has died.
/// </summary>
public sealed class BotDeathEvent : BotEvent
{
    /// <summary>The id of the bot that has died.</summary>
    public int VictimId { get; }

    /// <summary>Initializes a new instance of the BotDeathEvent class.</summary>
    /// <param name="turnNumber">Turn number when the bot died.</param>
    /// <param name="victimId">Id of the bot that has died.</param>
    [JsonConstructor]
    public BotDeathEvent(int turnNumber, int victimId) : base(turnNumber) => VictimId = victimId;
}